### PR TITLE
Glass tile stuff

### PR DESCRIPTION
**Pull request by Laboredih123**
## About the PR 
Adds large glass transparent tiles, adds a type of these tiles that can autoconnect, and glass tiles emit light.

## Why's this needed? 
It looks real nice.
image

## Changelog 

```changelog
(u)cringe
(+)Nanotrasen has built better glass floors that are seamless. They also let through light.
```

### CODE REVIEW

**Comment by itchybones** [comment]
Will these be added to the floor designer selection?

**Comment by Laboredih123** [comment]
idk

**Comment by Laboredih123** [comment]
should i add it?

**Comment by Laboredih123** [comment]
what happened to the checks what

File: code/turf/floors.dm [hunk 1]
```diff
@@ -2677,6 +2697,109 @@ TYPEINFO(/turf/simulated/floor/auto/water/ice)
 		if(prob(10))
 			src.icon_state = "snow_rough[rand(1,3)]"
 
+TYPEINFO(/turf/simulated/floor/auto/glassblock)
+	mat_appearances_to_ignore = list("steel","synthrubber","glass")
+	connect_diagonal = 1
+
+//dont use please
```
Comment (ZeWaka) [inline code/turf/floors.dm:2704]: Then abstract it?

**Comment by itchybones** [comment]
> should i add it?

I think so, yeah, players woulf come up with lots of cool ways of using these. Although this is entirely out of this PR's scope, it'd be neat for these to be craftable through some way.

**Comment by WALPVRGIS** [comment]
Do all glass floors emit light now? I feel like that will affect a bunch of mapping I've done :o would like to account for this

**Comment by Laboredih123** [comment]
> Do all glass floors emit light now (as of this PR anyway)? I feel like that will affect a bunch of mapping I've done :o would like to account for this

all of them yeah since it felt wrong for them to be dark, if you want me to remove that for some glass tiles like the nonlarge ones i suppose i can

File: code/turf/floors.dm (hunk 1 above)
Comment (Laboredih123) [reply code/turf/floors.dm:2704]: already tried, every map check proceeded to fail

Comment (Laboredih123) [reply code/turf/floors.dm:2704]: /turf/simulated/floor is a used type

File: code/turf/floors.dm [hunk 2]
```diff
@@ -2677,6 +2697,109 @@ TYPEINFO(/turf/simulated/floor/auto/water/ice)
...
+	to_plating()
+		return
+
+	break_tile_to_plating()
+		return
+
+	break_tile()
+		return
+
+	// unburnable, otherwise floorbots use steel sheets for repair which doesn't make sense
```
Comment (ZeWaka) [inline code/turf/floors.dm:2754]: we can revisit this if it's a problem i guess

**Review by ZeWaka — APPROVED** [review]

**Comment by Laboredih123** [comment]
oh huh

**Comment by Laboredih123** [comment]
nice

**Pull request merged**